### PR TITLE
Feature/planet travel gui

### LINKED ISSUE #11: UI Implementation for Planetary Display.
# **Description**
**Task:** UI Implementation for Planetary Display.
**Feature:** [Space Feature](#3)

The planetary display UI serves as a pivotal element in our space-themed game. It not only provides players with essential information about their current location but also offers an immersive and visually captivating experience.

[] Develop the user interface components to showcase both the current and future planets in the game as per the designed layout. 
[] Create the user interface components responsible for displaying details of the current planet in the game.
Ensure that the current planet's information is visually appealing and coherent with the game's theme.
Consider the importance of this feature in providing players with vital information about their current location and the immersive gaming experience it offers.

## **Example (optional)**
-Implement the UI for Displaying Current Planet
- **Part**: Implement the UI components responsible for showing current planet

# **Dependencies**
- [ ] #57 57

# **Milestones**
- [x] Create UI components for showing current planet details (Aug. 23).
- [x] unit test coverage of code (Aug. 30) :  Ensure comprehensive unit test coverage to validate the functionality of the UI components and interactions.
- [x] Documentation (Aug. 30)

**Completion Deadline:** Aug. 30

# **Documentation**
 Shaivika:
- (https://github.com/UQcsse3200/2023-studio-2/wiki/Navigation-Component)[ttps://github.com/UQcsse3200/2023-studio-2/wiki/Navigation-Component](https://github.com/UQcsse3200/2023-studio-2/wiki/Navigation-Component)

# **Member**

### LINKED ISSUE #57: UI Design for Planetary Display
# **Description**
**Task:** UI Design for Planetary Display.
**Feature:** [Space Feature](#3)

Design the user interface component to showcase both the current and future planets in the game. This involves creating a visually appealing layout, determining the placement of UI elements for displaying planet details, and conceptualising the interactive features for selecting future planets for travel.

## **Example (optional)**
- **Part**: Using a design tool, create a high-fidelity mockup of the ui showing the current planet

# **Dependencies**
- None

# **Milestones**
- [x] Design the UI layout for planetary display and finalize the visual concept
- [x] Design navigation elements,Ensure that navigation controls are intuitive and visually aligned with the overall theme.
- [x] Test the UI design with a sample group of players or users.
- [x] Finalize all design assets, including images, icons, and interactive elements.

**Completion Deadline:** 28Aug

# **Documentation**
Manandeep -
https://github.com/UQcsse3200/2023-studio-2/wiki/MAIN-MENU-AND-SETTING-MENU-DISPLAY-CHANGES

# **Member**

### PR DESCRIPTION
**Pull request by shaivikaaaa**
This pull request introduces a new feature of UI implementation of planetary display I'm which there is specific space button throughout which a user can access to different levels of the game

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/NavigationArea.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import com.badlogic.gdx.Gdx;
+import com.csse3200.game.GdxGame;
+import com.csse3200.game.areas.terrain.TerrainFactory;
+import com.csse3200.game.components.gamearea.GameAreaDisplay;
+import com.csse3200.game.components.navigation.NavigationPlanetComponent;
+import com.csse3200.game.entities.Entity;
+import com.csse3200.game.services.ResourceService;
+import com.csse3200.game.services.ServiceLocator;
+
+public class NavigationArea extends GameArea{
```
Comment (TDay1) [inline source/core/src/main/com/csse3200/game/areas/NavigationArea.java:18]: Class needs JavaDocs

File: source/core/src/main/com/csse3200/game/components/navigation/PlanetComponentt.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private final float x;
+    private final float y;
+    private Table table;
+    public PlanetComponentt(String PlanetImage,float x,float y)
+    {
+
+        this.PlanetImage=PlanetImage;
+        this.x=x;
+        this.y=y;
+
```
Comment (TDay1) [inline source/core/src/main/com/csse3200/game/components/navigation/PlanetComponentt.java]: Remove whitespace

File: source/core/src/main/com/csse3200/game/components/navigation/PlanetComponentt.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        table.setSize(20,20);
+        stage.addActor(table);
+    }
+
+
+
+    @Override
+    protected void draw(SpriteBatch batch) {
+
+
```
Comment (TDay1) [inline source/core/src/main/com/csse3200/game/components/navigation/PlanetComponentt.java]: Remove whitespace

File: source/core/src/main/com/csse3200/game/components/navigation/level4.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    private final float x;
+    private final float y;
+    private Table table;
+    public level4(String PlanetImage,float x,float y)
+    {
+
+        this.PlanetImage=PlanetImage;
+        this.x=x;
+        this.y=y;
+
```
Comment (TDay1) [inline source/core/src/main/com/csse3200/game/components/navigation/level4.java]: Remove whitespace

File: source/core/src/main/com/csse3200/game/components/navigation/level4.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        Button button=new Button(label,skin);
+        button.addListener(
+                new ChangeListener() {
+                    @Override
+                    public void changed(ChangeEvent changeEvent, Actor actor) {
+
+                        entity.getEvents().trigger("Navigateeee");
+                    }
+                });
+
```
Comment (TDay1) [inline source/core/src/main/com/csse3200/game/components/navigation/level4.java:61]: Whitespace

File: source/core/src/main/com/csse3200/game/components/navigation/level4.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        table.setPosition(x,y);
+        table.setSize(20,20);
+        stage.addActor(table);
+    }
+
+
+
+    @Override
+    protected void draw(SpriteBatch batch) {
+
```
Comment (TDay1) [inline source/core/src/main/com/csse3200/game/components/navigation/level4.java:77]: Whitespace

**Review by TDay1 — CHANGES_REQUESTED** [review]

**Review by TDay1 — APPROVED** [review]

**Review by henrybatt — APPROVED** [review]
Looks good

**Pull request merged**